dot/core/service.go

```diff
@@ -497,6 +497,10 @@ func (s *Service) HandleSubmittedExtrinsic(ext types.Extrinsic) error {
 		return nil
 	}
 
+	if s.transactionState.Exists(ext) {
```

Comment:
This function is just to avoid calling multiple times the validation function to the same transaction, but you're right, there is no chance to have any duplicated in the pool!

Reply:
Sorry, if I don't have the context here.

Do we have any logic because of which we might call this function multiple times for the same transaction? or are users of gossamers likely call this for the same transaction?

If that is not likely to happen, I would likely not have this check.